lib/gears/shape.lua

```diff
@@ -472,8 +472,11 @@ function module.arc(cr, width, height, thickness, start_angle, end_angle, start_
     if end_rounded then
         arc_length = arc_length - thickness/2
 
-        -- And back to angles
-        end_angle = start_angle + (arc_length/(radius - thickness/2))
+        -- And back to angles. Also make sure to avoid underflowing when the
+        -- rounded edge radius is greater than the angle delta.
+        end_angle = math.max(
+            start_angle, start_angle + (arc_length/(radius - thickness/2))
```

Comment:
Also, doesn't the subtraction around `start_angle` in line 469 need the same treatment?